src/roswire/definitions/package.py

```diff
@@ -94,17 +94,13 @@ def paths(shell: ShellProxy, file_proxy: FileProxy) -> List[str]:
         package_paths: List[str] = path_str.strip().split(':')
         paths: List[str] = []
         for path in package_paths:
-            if file_proxy.isfile(f'{path}/package.xml'):
-                paths.append(path)
-            else:
-                try:
-                    dirs: List[str] = file_proxy.listdir(path, absolute=True)
-                except OSError:
-                    # Not a directory
-                    continue
-                for d in dirs:
-                    if file_proxy.isfile(f'{d}/package.xml'):
-                        paths.append(d)
+            try:
+                all_packages = files.find(path, 'package.xml')
+            except OSError:
```

Comment:
https://docs.python.org/3/library/contextlib.html#contextlib.suppress

Reply:
But I don't want to suppress it. I want to continue the for loop if there's an exception.

Reply:
Ah. Good point.